src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -78,6 +79,17 @@ void recycle()
    {
       if (connection != null) {
          this.lastAccessed = currentTime();
+         try {
+            DatabaseMetaData dm = connection.getMetaData();
+            if (dm!=null){
+               if (dm.getJDBCMajorVersion() > 4 ||
+                  (dm.getJDBCMajorVersion() == 4 && dm.getJDBCMinorVersion() >= 3)) {
```

Comment:
This logic should not be executed every time. Can this be moved somewhere so the calculation of if the feature is supported is only run once. Calling getMetaData might not be cheap.

Reply:
This comment has been addressed in pull request https://github.com/brettwooldridge/HikariCP/pull/2126

Reply:
@lfbayer FYI I've asked Deeptanshu to close this PR as his internship at Oracle is now over. We will continue his work in PR [2126](https://github.com/brettwooldridge/HikariCP/pull/2126).